homeassistant/components/doorbird/__init__.py

```diff
@@ -18,45 +18,23 @@
 API_URL = '/api/{}'.format(DOMAIN)
 
 CONF_CUSTOM_URL = 'hass_url_override'
-CONF_DOORBELL_EVENTS = 'doorbell_events'
-CONF_DOORBELL_NUMS = 'doorbell_numbers'
-CONF_RELAY_NUMS = 'relay_numbers'
-CONF_MOTION_EVENTS = 'motion_events'
-
-SENSOR_TYPES = {
-    'doorbell': {
-        'name': 'Button',
-        'device_class': 'occupancy',
-    },
-    'motion': {
-        'name': 'Motion',
-        'device_class': 'motion',
-    },
-    'relay': {
-        'name': 'Relay',
-        'device_class': 'relay',
-    }
-}
+CONF_EVENTS = 'events'
 
 RESET_DEVICE_FAVORITES = 'doorbird_reset_favorites'
 
 DEVICE_SCHEMA = vol.Schema({
     vol.Required(CONF_HOST): cv.string,
     vol.Required(CONF_USERNAME): cv.string,
     vol.Required(CONF_PASSWORD): cv.string,
-    vol.Optional(CONF_DOORBELL_NUMS, default=[1]): vol.All(
-        cv.ensure_list, [cv.positive_int]),
-    vol.Optional(CONF_RELAY_NUMS, default=[1]): vol.All(
-        cv.ensure_list, [cv.positive_int]),
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Optional(CONF_EVENTS, default=[]): vol.All(
+        cv.ensure_list, [cv.string]),
```

Comment:
Is there no need to validate what events are available?

Reply:
The event name is used on the URL that is registered on the Doorbird device.  As long as it is a URL safe event name it should work fine.  We could validate for that if you think it is warranted.

Reply:
Is there a limited number of event names that we can check against?

Reply:
The Doorbird device doesn't care what the event is called.  We could artificially limit it, but that would require update to the component to support additional features.  This change was prompted by the release of units with multiple buttons.  Previously, we only allowed a single doorbell event to be registered and validated the name via `monitored_conditions` config.  Doorbird doesn't currently provide a way to programmatically identify the number of buttons a unit has.  This means it is easier to allow the user to specify as many custom events as needed and they can attach them to the necessary actions via the Doorbird app.  No changes required to maintain the possible event names when a new device is released or a new action is made available via software update.

Reply:
Ok.